### ISSUE TITLE
Issue with numpy 1.13.1

### ISSUE THREAD

**Issue by mkovalski** [inducer/pycuda#178]
Hello,

I'm trying to run a basic example of sparse matrix vector multiplication using PacketedSpMV, but running into an issue when attempting to run this. I'm attempting to replicate the performance found in [this stackoverflow post](https://stackoverflow.com/questions/49019189/how-can-i-accelerate-a-sparse-matrix-by-dense-vector-product-currently-implemen?utm_medium=organic&utm_source=google_rich_qa&utm_campaign=google_rich_qa)

```
import numpy as np
import pycuda.autoinit
import pycuda.driver as drv
import pycuda.gpuarray as gpuarray
from pycuda.sparse.packeted import PacketedSpMV
from pycuda.tools import DeviceMemoryPool
from scipy.sparse import csr_matrix

N = 1000
P = 0.1
DTYPE = np.float32

a_dense = np.random.rand(N, N).astype(DTYPE)
a_dense[np.random.rand(N, N) >= P] = 0
a_sparse = csr_matrix(a_dense)

spmv = PacketedSpMV(a_sparse, is_symmetric=False, dtype=dtype)
```

On execution of the PacketedSpMV line, I get the following error.

```
in pycuda.sparse.pkt_build_cython.build_pkt_data_structure (/home/mhk/.pyxbld/temp.linux-x86_64-2.7/pyrex/pycuda/sparse/pkt_build_cython.c:1219)()
     20     base_dof_nr = 0
     21
---> 22     index_array = numpy.zeros(
     23             max_thread_costs*thread_count, dtype=spmv.packed_index_dtype)
     24     data_array = numpy.zeros(


TypeError: 'numpy.float64' object cannot be interpreted as an index
```

I saw this issue somewhere else and saw a solution would be to downgrade to use numpy 1.11. Are there any plans to fix this so that this could work with a later version of numpy?

**Comment by inducer** [comment]
Should be fixed by 117afa6e863615145d11a37e047952cc0e26fe2d.

**Comment by inducer** [comment]
Thanks for the report.
